language/tools/move-package/src/resolution/resolution_graph.rs

```diff
@@ -526,8 +540,17 @@ impl ResolvingGraph {
         };
 
         for (dep_name, dep) in manifest.dependencies.iter().chain(additional_deps.iter()) {
-            Self::download_and_update_if_remote(*dep_name, dep)?;
+            if let Some(git_info) = &dep.git_info {
+                if !git_info.download_to.exists() {
```

Comment:
Consider also 
```
let already_downloaded = dep.git_info.map(|gi| gi.download_to.exists()).unwrap_or(true);
if !already_downloaded {
   Self::increase_movey_download_count( ... )
}
```

Reply:
This can be resolved depending on the solution of https://github.com/move-language/move/pull/318#discussion_r968187683